File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/core/models/rotorrouterwithleaderdispersion/RotorRouterWithLeaderRobot.java
package hu.elteik.knowledgelab.javaengine.core.models.rotorrouterwithleaderdispersion;

import hu.elteik.knowledgelab.javaengine.core.models.base.Color;
import hu.elteik.knowledgelab.javaengine.core.models.base.Robot;
import hu.elteik.knowledgelab.javaengine.core.models.base.RobotState;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RotorRouterWithLeaderRobot extends Robot {

    private Long lastUsedEdgeID;

    public RotorRouterWithLeaderRobot(Long ID, RobotState state, Color color, Long onID, Long destinationID, Long lastUsedEdgeID) {
        super(ID, state, color, onID, destinationID);
        this.lastUsedEdgeID = lastUsedEdgeID;
    }
}

File: dispersion-app/src/app/models/rotor-router-with-leader-dispersion-entities/RotorRouterWithLeaderRobot.ts
import {Robot, RobotState} from "../base-entities/Robot";

export class RotorRouterWithLeaderRobot extends Robot{

  lastEdgeID: number;

  constructor(id?: number, onID?: number, state?: RobotState, color?: string, destinationID?: number , lastEdgeID?: number) {
    super(id, onID, state, color, destinationID)
    this.lastEdgeID = lastEdgeID;
  }

}
